ui/selectors/approvals.ts

```diff
@@ -0,0 +1,19 @@
+import { ApprovalRequest } from '@metamask/approval-controller';
+import { ApprovalType, Json } from '@metamask/controller-utils';
+
+type ApprovalsMetaMaskState = {
+  metamask: {
+    pendingApprovals: ApprovalRequest<Record<string, Json> | null>[];
+  };
+};
+
+export function hasPendingApprovalsSelector(
+  state: ApprovalsMetaMaskState,
+  approvalType: ApprovalType,
+) {
+  const pendingApprovalRequests = Object.values(state.metamask.pendingApprovals)
```

Comment:
I was wondering if I had to check for undefined at some point here, but none of the references to this or many other state properties do across the code, so I'm assuming it's safe.